### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files to check
`src/App.js`
`src/components/ProtectedRoute.jsx`
`src/components/PublicRoute.js`
`src/components/navbar/Navbar.jsx`
`src/context/auth-context.js`
`src/context/toast-context.js`
`src/page/forms/Login.jsx`
`src/page/forms/Signup.jsx`
`src/page/profile/Profile.jsx`
`src/reducer/auth-reducer.js`

### Thank you for reviewing my pr

### CODE REVIEW

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -2,11 +2,18 @@ import styles from "./navbar.module.css";
...
+import { Link } from "react-router-dom";
 export const Navbar = () => {
   const { setActiveAside } = useAside();
   const clickHandler = () => {
     setActiveAside((status) => !status);
   };
+  const {
+    authState: { authToken },
+  } = useAuth();
+  console.log(authToken);
```
Comment (ApurvaSawant11) [inline src/components/navbar/Navbar.jsx]: Remove this if not needed

File: src/page/forms/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    //     ...prevVal,
+    //     bg: "red",
+    //     isOpen: true,
+    //     msg: "Password do not match",
+    //   }));
+    //   setTimeout(
+    //     setToastVal((prevVal) => ({ ...prevVal, isOpen: false })),
+    //     1500
+    //   );
+    //}
```
Comment (ApurvaSawant11) [inline src/page/forms/Signup.jsx:35]: Remove commented code if not needed

File: src/page/forms/form.module.css [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  transform: scale(1.02, 1.02);
+}
+.linkWrapper {
+  margin: 1rem 0;
+}
+.passInput {
+  position: relative;
+}
+.passInputWrapper {
+  position: relative;
```
Comment (ApurvaSawant11) [inline src/page/forms/form.module.css:40]: You can merge this `.passInput` `.passInputWrapper` if there is nothing extra that needs to be added.

File: src/page/forms/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          ) : (
+            <i
+              className="fa fa-eye-slash eye"
+              onClick={() => setType("password")}
+            ></i>
+          )}
+        </div>
+
+        <button className={styles.submitBtn} onClick={(e) => submitHandler(e)}>
+          Login with credentials
```
Comment (ApurvaSawant11) [inline src/page/forms/Login.jsx:56]: This won't work. You are not passing test credentials anywhere.

**Review by ApurvaSawant11 — APPROVED** [review]
Everything looks good. Didn't find much to suggest.

Just check if your login and signup are working properly or not for the **deployed site**.
I am getting errors while logging in or signup. And the same toast is displayed `error signing up`

File: src/components/navbar/Navbar.jsx (hunk 1 above)
Comment (Developers-codz) [reply src/components/navbar/Navbar.jsx]: will do that

File: src/page/forms/Signup.jsx (hunk 2 above)
Comment (Developers-codz) [reply src/page/forms/Signup.jsx:35]: will need that later

File: src/page/forms/form.module.css (hunk 3 above)
Comment (Developers-codz) [reply src/page/forms/form.module.css:40]: Will look into that

File: src/page/forms/Login.jsx (hunk 4 above)
Comment (Developers-codz) [reply src/page/forms/Login.jsx:56]: I am passing it in auth context in login function

**Comment by Developers-codz** [comment]
> Everything looks good. Didn't find much to suggest.
> 
> Just check if your login and signup are working properly or not for the **deployed site**. I am getting errors while logging in or signup. And the same toast is displayed `error signing up`

checked it just now its not working for deployed site only, Will resolve it thank you

**Comment by Developers-codz** [comment]
> > Everything looks good. Didn't find much to suggest.
> > Just check if your login and signup are working properly or not for the **deployed site**. I am getting errors while logging in or signup. And the same toast is displayed `error signing up`
> 
> checked it just now its not working for deployed site only, Will resolve it thank you

Its working now, you can check it

**Review by b0llu — APPROVED** [review]
Found nothing to suggest except the old suggestions. Everything else is looking good. Nice clean code

File: src/context/auth-context.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      setToastVal((prevVal) => ({
+        ...prevVal,
+        bg: "green",
+        msg: "Successfully Signed up",
+        isOpen: true,
+      }));
+      setTimeout(
+        () => setToastVal((prevVal) => ({ ...prevVal, isOpen: false })),
+        1500
+      );
```
Comment (partha8) [inline src/context/auth-context.js]: suggestion : you can keep this timeout in the toast component, just use useEffect and in the dependency have isOpen.

File: src/reducer/auth-reducer.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        email: email,
+        isAuthTokenPresent: true,
+        authToken: token,
+      };
```
Comment (partha8) [inline src/reducer/auth-reducer.js:40]: suggestion/can be done later: If you look closely, the two cases are almost the same, so you can have just one case for login/signup. Do play around with this.

**Review by partha8 — APPROVED** [review]
The code is clean, readable and the functions work nicely. Good work!

File: src/context/auth-context.js (hunk 5 above)
Comment (Developers-codz) [reply src/context/auth-context.js]: Done!!!

File: src/reducer/auth-reducer.js (hunk 6 above)
Comment (Developers-codz) [reply src/reducer/auth-reducer.js:40]: Okay

**Pull request merged**